packages/components/src/gridicon/index.tsx

```diff
@@ -17,11 +17,46 @@ interface Props {
 
 type AllProps = Assign< React.SVGProps< SVGSVGElement >, Props >;
 
+function isCrossOrigin( url: string ) {
+	try {
+		return new URL( url ).origin !== window.location.origin;
+	} catch ( e ) {
+		return false;
+	}
+}
+
+/**
+ * Fetches the SVG file and creates a proxy URL for it.
+ * @param url the original cross-origin URL
+ * @returns the proxied URL.
+ */
+function useProxiedURL( url: string ) {
+	const isCrossOriginUrl = isCrossOrigin( url );
+	// If cross-origin, return `undefined` until the proxy URL is fetched.
+	const [ urlProxy, setUrlProxy ] = React.useState( isCrossOriginUrl ? undefined : url );
```

Comment:
We can prevent calling `isCrossOriginUrl` on each render, where the result is rarely relevant:
```js
const [ urlProxy ] = useState( () => isCrossOrigin( url ) ? undefined : url );
useEffect( () => {
  if ( isCrossOrigin( url ) ) {
    setUrlProxy( undefined );
    fetch( url ).then( ... );
  } else {
    setUrlProxy( url );
  }
}, [ url ] );
```
This also fixes a bug when the `url` prop changes from cross-origin to same-origin URL. The new same-origin value wouldn't be used.